zk_prover/src/chips/range/tests.rs

```diff
@@ -0,0 +1,287 @@
+            AssignedCell<Fp, Fp>,
+            AssignedCell<Fp, Fp>,
+        ),
+        Error,
+    > {
+        layouter.assign_region(
+            || "initialize value and sum",
+            |mut region| {
+                self.config.selector.enable(&mut region, 0)?;
+
+                let a_cell =
+                    region.assign_advice(|| "a", self.config.col_a, 0, || Value::known(a))?;
+
+                let b_cell =
+                    region.assign_advice(|| "b", self.config.col_b, 0, || Value::known(b))?;
+
+                let c_cell = region.assign_advice(
+                    || "a + b",
+                    self.config.col_c,
+                    0,
+                    || a_cell.value().copied() + b_cell.value(),
+                )?;
+
+                Ok((a_cell, b_cell, c_cell))
+            },
+        )
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct TestConfig<const N_BYTES: usize> {
+    pub addchip_config: AddConfig,
+    pub range_check_config: RangeCheckConfig<N_BYTES>,
+}
+
+// The test circuit takes two inputs a and b.
+// It adds them together by using the add chip to produce c = a + b.
+// Performs a range check on a, b and c. Each value should lie in N_BYTES.
+#[derive(Default, Clone, Debug)]
+struct TestCircuit<const N_BYTES: usize> {
+    pub a: Fp,
+    pub b: Fp,
+}
+
+impl<const N_BYTES: usize> Circuit<Fp> for TestCircuit<N_BYTES> {
+    type Config = TestConfig<N_BYTES>;
+    type FloorPlanner = SimpleFloorPlanner;
+
+    fn without_witnesses(&self) -> Self {
+        Self::default()
+    }
+
+    fn configure(meta: &mut ConstraintSystem<Fp>) -> Self::Config {
+        let z = meta.advice_column();
+        let range = meta.fixed_column();
+
+        let a = meta.advice_column();
+        let b = meta.advice_column();
+        let c = meta.advice_column();
+
+        meta.enable_equality(z);
+        meta.enable_equality(a);
+        meta.enable_equality(b);
+        meta.enable_equality(c);
+
+        let constants = meta.fixed_column();
+        meta.enable_constant(constants);
+
+        let add_selector = meta.selector();
+        let toggle_lookup_check = meta.complex_selector();
+
+        let range_check_config =
+            RangeCheckChip::<N_BYTES>::configure(meta, z, range, toggle_lookup_check);
+
+        let addchip_config = AddChip::configure(meta, a, b, c, add_selector);
+
+        {
+            TestConfig {
+                addchip_config,
+                range_check_config,
+            }
+        }
+    }
+
+    fn synthesize(
+        &self,
+        config: Self::Config,
+        mut layouter: impl Layouter<Fp>,
+    ) -> Result<(), Error> {
+        // Initiate the add chip
+        let addchip = AddChip::construct(config.addchip_config);
+        let (a_cell, b_cell, c_cell) =
+            addchip.assign(self.a, self.b, layouter.namespace(|| "add chip"))?;
+
+        // Initiate the range check chip
+        let overflow_chip = RangeCheckChip::construct(config.range_check_config);
```

Comment:
How about just use `range_chip` instead of `overflow_chip`?

Reply:
Yeah, that was a leftover from the previous version. Fixed in 969ac61